InnerEye/ML/common.py

```diff
@@ -85,34 +86,32 @@ def get_best_checkpoint_path(path: Path) -> Path:
     return path / BEST_CHECKPOINT_FILE_NAME_WITH_SUFFIX
 
 
-def keep_latest(path: Path, search_pattern: str) -> Optional[Path]:
+def find_recovery_checkpoint_and_epoch(path: Path) -> Optional[Tuple[Path, int]]:
     """
-    Looks at all files that match the given pattern via "glob", and deletes all of them apart from the most most
-    recent file. The surviving file is returned. If there is no single file that matches the search pattern, then
-    return None.
+    Looks at all the recovery files, extracts the epoch number for all of them and returns the most recent (latest
+    epoch)
+    checkpoint path along with the corresponding epoch number. If no recovery checkpoint are found, return None.
     :param path: The folder to start searching in.
-    :param search_pattern: The glob pattern that specifies the files that should be searched.
-    :return: None if there is no file matching the search pattern, or a Path object that has the latest file matching
-    the pattern.
+    :return: None if there is no file matching the search pattern, or a Tuple with Path object and integer pointing to
+    recovery checkpoint path and recovery epoch.
     """
-    files_and_mod_time = [(f, f.stat().st_mtime) for f in path.glob(search_pattern)]
-    files_and_mod_time.sort(key=lambda f: f[1], reverse=True)
-    for (f, _) in files_and_mod_time[1:]:
-        logging.info(f"Removing file: {f}")
-        f.unlink()
-    if files_and_mod_time:
-        return files_and_mod_time[0][0]
-    return None
+    filenames = [f for f in path.glob(RECOVERY_CHECKPOINT_FILE_NAME + "*")]
```

Comment:
for testability, it would be better to split the "glob" part and the regex-part into to functions.

Reply:
done and added two tests